ElrondNetwork/elrond-go
# EN-13328: overridable config fields

## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- During network upgrades, the entire `config.toml` should be replaced. This could create issues for node operators that perform changes to the `config.toml`, for example setting the `DbLookupExtension` to `true`. 
  
## Proposed Changes
- Added a section in `prefs.toml` where new values can be updated by simply specifying the path and the new value

## Testing procedure
- try to update as many paths as possible, alongside invalid combinations that should error
